server/emails/templates/BaseEmail.tsx

```diff
@@ -80,6 +81,21 @@ export default abstract class BaseEmail<T extends EmailProps, S = any> {
       Metrics.increment("email.sent", {
         templateName,
       });
+
+      // @ts-expect-error does not exist on type T & S
```

Comment:
This is unfortunate, the types have been tricky here with it being static. Perhaps we can define S as an object which always has an optional `notificationId`?

Reply:
Added on both `T` and `S`.

Reply:
What is we add it `T` instead? `S = Optional<...>` thingy doesn't work since `S` would still be _unknown_. On the other hand, `S extends...` makes it _required_ which in result demands modifications at other places too.

Reply:
Or how about having and optional `metadata` field(`metadata?: Required<{ notificationId: string }>;`) in top level `EmailProps` instead? I guess that's better.